src/nodeViews/createReactNodeViewConstructor.tsx

```diff
@@ -67,6 +76,58 @@ export type ReactNodeViewConstructor = (
   ...args: Parameters<NodeViewConstructor>
 ) => ReactNodeView;
 
+const REACT_PM_KEY = "reactpmkey";
+
+function findNearestRegistryKey(editorView: EditorView, pos: number) {
+  let parentElement: HTMLElement | null = editorView.domAtPos(pos, 0)
+    .node as HTMLElement;
+  while (parentElement && !parentElement.dataset[REACT_PM_KEY]) {
+    const grandParentElement: HTMLElement | null = parentElement.parentElement;
+    parentElement = grandParentElement;
+  }
+  return parentElement?.dataset[REACT_PM_KEY] ?? PORTAL_REGISTRY_ROOT_KEY;
+}
+
+/** A queue of registration tasks to be executed */
+const taskQueue: Array<() => void> = [];
+
+/**
+ * Flush the task queue, executing all tasks
+ */
+function flushTaskQueue() {
+  const { length } = taskQueue;
```

Comment:
Should we just splice and then iterate over the elements of the result rather than removing them one-by-one?

Reply:
As long as we know that registration can't trigger cancellation. 🤯